utils/query-state/Get-UpstreamBranchMap.psm1

```diff
@@ -0,0 +1,28 @@
+function Get-UpstreamBranchMap([Array]$branches) {
+	$upstreams = @{};
+	$files = git ls-tree -r origin/_upstream --format="%(objectname)`t%(path)" `
+		| ForEach-Object {
+			$row = $_ -split "`t"
+			[psobject]@{ 
+				oid=$row[0];
+				path=$row[1] 
+			}
+		} `
+		| Where-Object { 
+			$branches -contains $_.path -or $branches.Length -eq 0 
+		};
+
+	$blobs = @{}
+	(git ls-tree -r origin/_upstream --format="%(objectname)" 
+		| git cat-file --batch="%(objectname) %(rest)")  -join "`n" -split "\n\n" `
```

Comment:
This is really cool, though it needs to leverage `Get-Configuration` to work in all the standard scenarios. This can be incorporated, for sure.

We can probably simplify this entire function a bit in order to only call the `git ls-tree` once... 🤔